File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int goodNodes(TreeNode root) {
        return count(root,root.val);
    }
    public int count(TreeNode root,int max){
     if(root==null) return 0;

     int pick = 0;
     if(root.val>=max){
        pick = 1;
        max = root.val;
     }

     pick+=count(root.left,max);
     pick+=count(root.right,max);
     return pick;
    }
}

File: Data Structures & Algorithms/max-area-of-island/submission-0.cpp
class Solution {
public:
    int maxAreaOfIsland(vector<vector<int>>& grid) {
      int n = grid.size();
      int m = grid[0].size();
      int ans = 0;
        for(int i = 0;i<n;i++){
          for(int j = 0;j<m;j++){
            if(grid[i][j]==1){
              int count = 0;
              dfs(i,j,grid,count);
              ans = max(ans,count);
            }
          }
        }
        return ans;


    }

    void dfs(int i, int j , vector<vector<int>> &grid,int &count){

      if(i<0 || j<0 || i>=grid.size() || j>=grid[0].size() || grid[i][j]==0){
        return;
      }

      grid[i][j] = 0;
      count++;
      dfs(i,j+1,grid,count);
      dfs(i,j-1,grid,count);
      dfs(i+1,j,grid,count);
      dfs(i-1,j,grid,count);
    }
};
